src/v/storage/readers_cache.cc

```diff
@@ -80,6 +80,19 @@ readers_cache::put(std::unique_ptr<log_reader> reader) {
     return ptr->make_cached_reader(this);
 }
 
+bool readers_cache::is_within_locked_range(
```

Comment:
nit: this is called `within` but would it be more accurate to say `intersects`? presumably the range does not need to be fully contained, but merely overlap?

Reply:
exactly, will change the name